adapters/handlers/grpc/v1/batch_parse_request.go

```diff
@@ -45,7 +45,7 @@ func BatchFromProto(req *pb.BatchObjectsRequest, authorizedGetClass func(string,
 
 		class, err := authorizedGetClass(obj.Collection, obj.Tenant)
 		if err != nil {
-			objectErrors[insertCounter] = err
+			objectErrors[i] = err
```

Comment:
why this change? We probably used insertCounter for a reason?

Reply:
This caused a bug when `authorizedGetClass` fails for each of the objects within the batch. In such a case, `indexCounter` would never be incremented since the L49 `continue` would go to the next iteration of the loop. As such, `objectErrors[0]` would be overwritten each time leading to `len(objectErrors) == 1` regardless of `len(objectsBatch)`

I made this change to align it with how the other errors in the loop are handled, e.g. `L66`, `L70`, `L77`, `L100`. If we want to use `insertCounter` here instead of `i` then we need to increment `insertCounter` inside each error handling block. wdyt?

Reply:
I see, makes sense!